trustyai-service-operator/README.md

```diff
@@ -0,0 +1,141 @@
+        spec:
+          containers:
+            - name: trustyai-operator
+              image: quay.io/trustyai/trustyai-service-operator:latest
+              command:
+                - /manager
+              resources:
+                limits:
+                  cpu: 100m
+                  memory: 30Mi
+                requests:
+                  cpu: 100m
+                  memory: 20Mi
+    ```
+
+   or run
+
+   ```shell
+   kubectl apply -f https://raw.githubusercontent.com/trustyai-explainability/trustyai-service-operator/main/artifacts/examples/deploy-operator.yaml   
+   ```
+
+## Usage
+
+Once the operator is installed, you can create `TrustyAIService` resources, and the operator will create corresponding TrustyAI deployments, services, and (on OpenShift) routes.
+
+Here's an example `TrustyAIService` manifest:
+
+```yaml
+apiVersion: trustyai.opendatahub.io.trusty.opendatahub.io/v1
+kind: TrustyAIService
+metadata:
+  name: trustyai-service-example
+spec:
+  storage:
+    format: "PVC"
+    folder: "/inputs"
+    size: "1Gi"
+  data:
+    filename: "data.csv"
+    format: "CSV"
+  metrics:
+    schedule: "5s"
+    batchSize: 5000 # Optional, defaults to 5000
+```
+
+You can apply this manifest with 
+
+```shell
+kubectl apply -f <file-name.yaml> -n $NAMESPACE
+```
+to create a service, where `$NAMESPACE` is the namespace where you want to deploy it.
+
+
+Additionally, in that namespace:
+
+* a `ServiceMonitor` will be created to allow Prometheus to scrape metrics from the service.
+* (if on OpenShift) a `Route` will be created to allow external access to the service.
+
+### Custom Image Configuration using ConfigMap
+
+You can configure the operator to use custom images by creating a `ConfigMap` in the operator's namespace. 
+The operator only checks the ConfigMap at deployment, so changes made afterward won't trigger a redeployment of services.
+
+Here's an example of a ConfigMap that specifies a custom image:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: trustyai-service-operator-config
+data:
+  trustyaiServiceImageName: 'quay.io/mycustomrepo/mycustomimage'
+  trustyaiServiceImageTag: 'v1.0.0'
+```
+
+You can apply this manifest with the following command, replacing `<file-name.yaml>` with the name of your manifest file:
+
+```shell
+kubectl apply -f <file-name.yaml> -n $OPERATOR_NAMESPACE
+```
+
+Please ensure the namespace specified is the same as the namespace where you have deployed the operator.
+
+After the ConfigMap is applied, you can then proceed to deploy the TrustyAI service.
+The operator will use the image name and tag specified in the ConfigMap for the deployment.
+
+If you want to use a different image or tag in the future, you'll need to update the ConfigMap and redeploy the operator to have the changes take effect. The running TrustyAI services won't be redeployed automatically. To use the new image or tag, you'll need to delete and recreate the TrustyAIService resources.
+
+## Contributing
```

Comment:
These two links will be broken. I would link to upstream `trustyai-service-operator`s repos files.

Reply:
Done